modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main/categorization/vocabularies/EditVocabulary.tsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import {ClayVerticalNav} from '@clayui/nav';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import {navigate, sub} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+import VocabularyService from '../services/VocabularyService';
+import {AssetType} from '../types/AssetType';
+import {IVocabulary} from '../types/IVocabulary';
+import EditAssociatedAssetTypes from './EditAssociatedAssetTypes';
+import EditGeneralInfo from './EditGeneralInfo';
+
+export default function EditVocabulary({
+	assetTypes,
+	backURL,
+	defaultLanguageId,
+	locales,
+	siteId,
+	spritemap,
+	vocabulary,
+}: {
+	assetTypes: AssetType[];
+	backURL: string;
+	defaultLanguageId: string;
+	locales: any[];
+	siteId: number;
+	spritemap: string;
+	vocabulary: IVocabulary;
+}) {
+	const [activeVerticalNavKey, setActiveVerticalNavKey] = useState('general');
+	const [initialItemData, setInitialItemData] = useState<IVocabulary>(
+		vocabulary
+			? vocabulary
+			: {
+					description: '',
+					name: '',
+					name_i18n: {
+						[defaultLanguageId]: '',
+					},
+				}
+	);
+	const [nameInputError, setNameInputError] = useState<string>('');
+
+	const handleVerticalNavChange = (verticalNav: string) => {
+		setActiveVerticalNavKey(verticalNav);
+	};
+
+	const verifyInputs = () => {
+		if (nameInputError || initialItemData.name === '') {
+			setNameInputError(
+				sub(
+					Liferay.Language.get('the-x-field-is-required'),
+					Liferay.Language.get('name')
+				)
+			);
```

Comment:
I notice it checks for the presence of `nameInputError` but the next step sets the `nameInputError`, is it supposed to be `vocabulary.name`?

Reply:
I add `nameInputError` in the check because I want it to return false in that case. 
But setting `nameInputError` is for the case of `vocabulary.name` (initialItemData.name) is empty. This happens when the name field never gets set at all so `nameInputError` stays in it's initial value of an empty string.

Should I add an "if" statement for the name before setting `nameInputError`?

Reply:
I can see the second case, the `nameInputError` could be empty in the very beginning when there's no name either, and it would go ahead and save.... maybe it does not need to be changed then, if the logic worked correctly already.